tests/Unit/DataStructures/Tensor/Expressions/Test_MixedOperations.cpp

```diff
@@ -293,12 +428,428 @@ void test_case3(const DataType& used_for_size,
+  using d_pi_type = d_pi_type<DataType, Dim>;
+
+  std::uniform_real_distribution<> distribution(0.1, 1.0);
+
+  // Not inputs to the eq, but needed to make some variables
+  // that are computed by raising something in order for symmetric
+  // assumptions to hold
+  const auto inverse_spacetime_metric =
+      make_with_random_values<tnsr::AA<DataType, Dim>>(generator, distribution,
+                                                       used_for_size);
+  const auto phi = make_with_random_values<tnsr::iaa<DataType, Dim>>(
+      generator, distribution, used_for_size);
+
+  // RHS: spacetime_deriv_gauge_function
+  const spacetime_deriv_gauge_function_type spacetime_deriv_gauge_function =
+      make_with_random_values<spacetime_deriv_gauge_function_type>(
+          generator, distribution, used_for_size);
+
+  // RHS: pi_two_normals
+  const pi_two_normals_type pi_two_normals =
+      make_with_random_values<pi_two_normals_type>(generator, distribution,
+                                                   used_for_size);
+
+  // RHS: pi
+  const pi_type pi =
+      make_with_random_values<pi_type>(generator, distribution, used_for_size);
+
+  // RHS: gamma0
+  const gamma0_type gamma0 = make_with_random_values<gamma0_type>(
+      generator, distribution, used_for_size);
+
+  // RHS: normal_spacetime_one_form
+  const normal_spacetime_one_form_type normal_spacetime_one_form =
+      make_with_random_values<normal_spacetime_one_form_type>(
+          generator, distribution, used_for_size);
+
+  // RHS: gauge_constraint
+  const gauge_constraint_type gauge_constraint =
+      make_with_random_values<gauge_constraint_type>(generator, distribution,
+                                                     used_for_size);
+
+  // RHS: spacetime_metric
+  const spacetime_metric_type spacetime_metric =
+      make_with_random_values<spacetime_metric_type>(generator, distribution,
+                                                     used_for_size);
+
+  // RHS: normal_dot_gauge_constraint
+  const normal_dot_gauge_constraint_type normal_dot_gauge_constraint =
+      make_with_random_values<normal_dot_gauge_constraint_type>(
+          generator, distribution, used_for_size);
+
+  // RHS: christoffel_second_kind
+  const christoffel_second_kind_type christoffel_second_kind =
+      make_with_random_values<christoffel_second_kind_type>(
+          generator, distribution, used_for_size);
+
+  // RHS: gauge_function
+  const gauge_function_type gauge_function =
+      make_with_random_values<gauge_function_type>(generator, distribution,
+                                                   used_for_size);
+
+  // RHS: pi_2_up
+  pi_2_up_type pi_2_up(used_for_size);
+  for (size_t a = 0; a < Dim + 1; a++) {
+    for (size_t b = 0; b < Dim + 1; b++) {
+      pi_2_up.get(a, b) = 0.0;
+      for (size_t c = 0; c < Dim + 1; c++) {
+        pi_2_up.get(a, b) += inverse_spacetime_metric.get(c, b) * pi.get(a, c);
+      }
+    }
```

Comment:
why not use TE here?

Reply:
In unit tests that test TEs, I avoid using them in computing inputs and expected results since TEs are the thing being tested